src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+                    md = self._metadata.copy()
+                    md[model.MD_DWELL_TIME] = self.scanner.dwellTime.value
+                    md[model.MD_ACQ_DATE] = self._tStart
+                    da = model.DataArray([[0]], md)
+                    self.data.notify(da)
+
+            else:
+                # All other status types denote errors. 
+                logging.error("Error %d: %s", err_status, ERRCODE_DICT[err_status])
+                self.state._set_value(HwError(ERRCODE_DICT[err_status]), force_write=True)
+                self._measuring_flag.clear()
+
+        elif isinstance(decoded_msg, Message_DATAFRAME_SRVNACK):
+            # A measurement data frame was received from the server
+            # Do not respond to this type.
+            if not self._measuring_flag.is_set():
+                # We should not receive this message if not measuring
+                raise HwError("Measurement data frame received while not measuring. ")
+
+            # convert the OptionalDataRecord type to a regular python dictionary for logging
+            data = {r.name : r.data for r in decoded_msg.odOptional.values()}
+            logging.info("Acq: %s", data)
+
+            # update metadata and vigilant attributes based on data coming back
+            # from the server
+            if "ServerVersion" in data:
+                self._metadata[model.MD_SW_VERSION] = data['ServerVersion']
+            if "ResultingFilename" in data:
+                self.scanner.filename.value = data['ResultingFilename']
+            if "ResultingGroupname" in data:
+                self.scanner.directory.value = data['ResultingGroupname']
+
+        elif isinstance(decoded_msg, Message_ENCODED_STATUSREPLY):
+            # Do not answer these types of messages
+            # Log the updated status from the server
+            err_status = decoded_msg.ecStatus
+            logging.debug("Status: 0x%x: %s", err_status, ERRCODE_DICT[err_status])
+
+            if err_status == PQ_STOPREASON_CODE_FINISHED_OK:
+                if self.isMeasuring():
+                    logging.info("Finished ok.")
+                    self._measuring_flag.clear()
+            elif err_status < 0:
+                logging.error("Error %d: %s", err_status, ERRCODE_DICT[err_status])
+                self.state._set_value(HwError(ERRCODE_DICT[err_status]), force_write=True)
+                self._measuring_flag.clear()
+                raise HwError("Error %d: %s" % (err_status, ERRCODE_DICT[err_status]))
+
+        elif isinstance(decoded_msg, Message_DATAFRAME_SRVREPLY):
+            # Do not answer these types of messages
+            # Log the updated status from the server
+            err_status = decoded_msg.ecStatus
+            logging.debug("Status Reply: 0x%x: %s", err_status, ERRCODE_DICT[err_status])
+
+            if err_status == PQ_STOPREASON_CODE_CONTINUE_OK:
+                if self.isMeasuring() and self._user_break.is_set():
```

Comment:
What if it's not measuring or no user_break flag? Shouldn't measuring_flag always been `clear()`'ed ?

Reply:
The problem is that this message also gets sent in some other circumstances (like when you first connect), so I dind't want to alter the flags unless it was in the measuring state.